src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -378,6 +381,38 @@ int ZYNQShim::xclSyncBO(unsigned int boHandle, xclBOSyncDirection dir, size_t si
   return ioctl(mKernelFD, DRM_IOCTL_ZOCL_SYNC_BO, &syncInfo);
 }
 
+int ZYNQShim::xclCopyBO(unsigned int dst_boHandle, unsigned int src_boHandle, size_t size,
+                        size_t dst_offset, size_t src_offset)
+{
+  int ret = -EOPNOTSUPP;
+#ifdef __aarch64__
+    std::pair<const unsigned int, ert_start_copybo_cmd *const> bo = mCmdBOCache->alloc<ert_start_copybo_cmd>();
+    ert_fill_copybo_cmd(bo.second, src_boHandle, dst_boHandle,
+                        src_offset, dst_offset, size);
+
+    int ret = xclExecBuf(bo.first);
+    if (ret == 0)
+        while (xclExecWait(1000) == 0);   if (ret) {
```

Comment:
put if (ret) into next line, pls.
duplicated loop doing xclExecWait(1000) of the one at line 400?

Reply:
Good catch -- obsolete old code; cleaned.